contracts/NonfungiblePositionManagerV4.sol

```diff
@@ -0,0 +1,201 @@
+        return (
+            position.nonce,
+            position.operator,
+            poolKey.currency0,
+            poolKey.currency1,
+            poolKey.fee,
+            position.tickLower,
+            position.tickUpper,
+            position.liquidity,
+            position.feeGrowthInside0LastX128,
+            position.feeGrowthInside1LastX128,
+            position.tokensOwed0,
+            position.tokensOwed1
+        );
+    }
+
+    /// @inheritdoc INonfungiblePositionManagerV4
+    function mint(MintParams calldata params)
+        external
+        payable
+        override
+        checkDeadline(params.deadline)
+        returns (uint256 tokenId, uint128 liquidity, uint256 amount0, uint256 amount1)
+    {
+        // TODO: implement this
+    }
+
+    /// @inheritdoc INonfungiblePositionManagerV4
+    function increaseLiquidity(IncreaseLiquidityParams calldata params)
+        external
+        payable
+        override
+        checkDeadline(params.deadline)
+        returns (uint128 liquidity, uint256 amount0, uint256 amount1)
+    {
+        // TODO: implement this
+    }
+
+    /// @inheritdoc INonfungiblePositionManagerV4
+    function decreaseLiquidity(DecreaseLiquidityParams calldata params)
+        external
+        payable
+        override
+        isAuthorizedForToken(params.tokenId)
+        checkDeadline(params.deadline)
+        returns (uint256 amount0, uint256 amount1)
+    {
+        // TODO: implement this
+    }
+
+    /// @inheritdoc INonfungiblePositionManagerV4
+    function collect(CollectParams calldata params)
+        external
+        payable
+        override
+        isAuthorizedForToken(params.tokenId)
+        returns (uint256 amount0, uint256 amount1)
+    {
+        // TODO: implement this
+    }
+
+    /// @inheritdoc INonfungiblePositionManagerV4
+    function burn(uint256 tokenId) external payable override isAuthorizedForToken(tokenId) {
+        TokenPosition storage position = _positions[tokenId];
+        if (position.liquidity != 0 || position.tokensOwed0 != 0 || position.tokensOwed1 != 0) revert NotCleared();
+        delete _positions[tokenId];
+        _burn(tokenId);
+    }
+
+    modifier isAuthorizedForToken(uint256 tokenId) {
+        if (!_isApprovedOrOwner(msg.sender, tokenId)) revert NotApproved();
+        _;
+    }
+
+    function tokenURI(uint256 tokenId) public view override(ERC721, IERC721Metadata) returns (string memory) {
+        // TODO: implement this
+    }
+
+    /// @inheritdoc IERC721
+    function getApproved(uint256 tokenId) public view override(ERC721, IERC721) returns (address) {
+        if (!_exists(tokenId)) revert NonexistentToken();
+
+        return _positions[tokenId].operator;
+    }
+
+    /// @dev Overrides _approve to use the operator in the position, which is packed with the position permit nonce
+    function _approve(address to, uint256 tokenId) internal override(ERC721) {
```

Comment:
hmmm I'm kinda confused with the approval/auth scheme we are defining. It sounds like we are trying to override the traditional 721 spec but we are still using(not dis-allowing) the old 721 interfaces.

and we are using the old traditional 721 checks like `_isApprovedOrOwner` which doesnt look at the operator in the TokenPositions struct

Reply:
Also it seems like we want to allow permits... but where is all that sig verification?